src/components/ingestion/HttpIngestionEngine.js

```diff
@@ -0,0 +1,53 @@
+import { PassThrough } from 'stream'
+import fetch from 'node-fetch'
+import AbstractIngestionEngine from '%src/components/ingestion/AbstractIngestionEngine'
+
+/**
+ * The HttpIngestionEngine handles processing a HTTP video stream. It is a concrete
+ * implementation of AbstractIngestionEngine, which reads in a stream,
+ * converts it into an MPEG-TS stream represented as a series of Payloads,
+ * and then pipes that to Kafka.
+ *
+ * @extends AbstractIngestionEngine
+ */
+class HttpIngestionEngine extends AbstractIngestionEngine {
+	// The URL of the stream to read by the engine
+	url = null
+
+	/**
+	 * Create a HttpIngestionEngine.
+	 *
+	 * @param  {string}  url The url of the stream to be ingested
+	 */
+	constructor(url) {
+		if (!url) {
+			throw new Error(
+				'HttpIngestionEngine must be instantiated with a stream URL',
+			)
+		}
+		super()
```

Comment:
In Java I believe you *must* call super first; I appreciate that we don't need to reference `this` in our validation here, and so validation first works, but in the general case it's *possible* a validation rule might rely on the property of an inherited class and for that reason maybe standardizing around calling `super()` right away is worthwhile.

(if you disagree, plz disregard this comment!)

Reply:
Changed.